pkg/network/vmicontroller/status.go

```diff
@@ -124,3 +128,11 @@ func calculateSecondaryIfaceStatuses(
 
 	return interfaceStatuses, nil
 }
+
+func filterGuestInternalInterfaces(ifaceStatuses []v1.VirtualMachineInstanceNetworkInterface) []v1.VirtualMachineInstanceNetworkInterface {
+	return vmispec.FilterInterfacesStatus(
+		ifaceStatuses,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
+			return ifaceStatus.Name == "" && vmispec.ContainsInfoSource(ifaceStatus.InfoSource, vmispec.InfoSourceGuestAgent)
```

Comment:
I find this hard to understand.

I understand that in theses cases, we should preserve the entries:
- The infoSource has only `InfoSourceGuestAgent`.
- The infoSource has only `InfoSourceDomain`.
- The infoSource has only `InfoSourceGuestAgent` and `InfoSourceGuestAgent`.

The above should also imply that it is already in the already-calculated interfaces.
I would prefer to see it is indeed the case by expressing it in the unit-tests. It may be a bit tricky.
(I mean, how we are sure it cannot happen)

I am unsure about the `Name` relevance, it seems like a dependency on some side-effect. If that is the case, I consider the logic fragile.